### ISSUE TITLE
Add NK_USE_STD_MEM/MATH macros for NK_MEMSET/NK_SQRT series of functions?

### ISSUE THREAD

**Issue by YukinoHayakawa** [vurtun/nuklear#736]
The comment in the codes says I can define NK_MEMSET as memset, NK_MEMCPY as memcpy, NK_SQRT as sqrt, NK_SIN as sinf......etc if I want to use the std library versions of those functions. I think a bigger switch that enables using std version for a group of functions like NK_USE_STD_MEM for both NK_MEMSET  and NK_MEMCPY and NK_USE_STD_MATH for NK_SQRT, NK_SIN, NK_COS, etc would be more convenient.

**Comment by dumblob** [comment]
What would be the use case then?

While reading such code I wouldn't know what it is doing and would prefer having 4 more lines while seeing exactly how is each macro defined.

Currently adding these macros sounds like overengineering to me (Nuklear is a single header library - not an integrated solution).

**Comment by YukinoHayakawa** [comment]
For convenience and consistency. It quite surprised me that we have 
```c++
NK_INCLUDE_FIXED_TYPES
NK_INCLUDE_DEFAULT_ALLOCATOR
NK_INCLUDE_STANDARD_IO
NK_INCLUDE_STANDARD_VARARGS
```
but no such equivalent for math functions.
I suspect most users of the library will have access to the standard library and if they are using the flags above they will likely want to do the same for sin/cos/sqrt and then they will find what? no such macro? That's the opposite of [Principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

Whatever you define, the code aren't any clearer. Either NK_INCLUDE_STANDARD_MATH or NK_SQRT the user probably have to refer to the documentation to reveal its purpose. If someone read `#define NK_SQRT sqrt` and think ok it uses sqrt somewhere and don't know what happens if he undefines it, it means he just does not care.

If adding such a macro can be interpreted as "overengineering", I'm afraid all compilation controlling macros can be thought that way.

**Comment by dumblob** [comment]
Fair enough. You got me :wink:. I have to admit, that the story of other `NK_INCLUDE_...` macros you've mentioned is not any more minimal as initially wanted and therefore you're totally right the proposed macro `NK_INCLUDE_STANDARD_MATH` makes sense to me.

Just issue a pull request and I'll merge it. Cheers.

**Comment by YukinoHayakawa** [comment]
Ok, maybe later. I need some time to get familiar with the codebase first 😅.
